rpm_ostree: add basic client-actor for upgrades

lucab opened the pull request:
This adds an initial actor for rpm-ostree CLI client.
It also sketches the first step in downloading and applying updates.

lucab commented on src/rpm_ostree/cli_upgrade.rs:
```diff
@@ -0,0 +1,23 @@
+        .arg("--lock-finalization")
+        .arg(release.checksum.to_string())
+        .output()
+        .with_context(|e| format_err!("failed to run rpm-ostree: {}", e))?;
```
@jlebon my understanding is that this is a blocking command (i.e. its exit code is reliable to signal success). If not, I may need to tweak the state machine a bit.

jlebon commented on src/rpm_ostree/cli_upgrade.rs:
```diff
@@ -0,0 +1,23 @@
+/// Upgrade and leave the new deployment locked.
+pub fn locked_upgrade(release: Release) -> Fallible<Release> {
+    let cmd = std::process::Command::new("rpm-ostree")
+        .arg("upgrade")
```
We just chatted about this OOB already, but just making a note here: this will have to change to `deploy` and the argument you give it would be `revision=$checksum`. Will fix up rpm-ostree.

jlebon replied:
https://github.com/projectatomic/rpm-ostree/pull/1846

lucab commented on src/rpm_ostree/cli_status.rs:
```diff
@@ -0,0 +1,59 @@
+#[derive(Debug, Deserialize)]
+pub struct DeploymentJSON {
+    booted: bool,
+    checksum: String,
```
Great catch, I was vaguely aware of this nuance but I didn't realized it would be relevant here. I'm fine with keeping the current behavior of rpm-ostree.

For reference, this specific codepath is not currently used. But I think future logic will depend on introspection, so I added this preemptively.

lucab commented:
@jlebon thanks a lot for the insightful review. I've opened a bunch of followup tickets for the things discussed offline, and addressed all code comments.
